File: LoggerConfigParser/src/main/scala/pl/otros/logview/logppattern/LogbackLayoutEncoderConverter.scala
package pl.otros.logview.logppattern

import java.util.Properties

import pl.otros.logview.api.LayoutEncoderConverter

class LogbackLayoutEncoderConverter extends LayoutEncoderConverter {
  override def convert(layoutPattern: String): Properties = {
    val result = LogbackPatternConverter.convert(layoutPattern)
    result match {
      case Right(map) =>
        val prop = new Properties()
        map.foreach { kv => prop.setProperty(kv._1, kv._2) }
        prop
      case Left(error) => throw new Exception(error)
    }
  }
}

File: olv-parser/src/main/java/pl/otros/logview/logppattern/LogbackLayoutEncoderConverter.java
package pl.otros.logview.logppattern;

import pl.otros.logview.api.LayoutEncoderConverter;

import java.util.Properties;

public class LogbackLayoutEncoderConverter implements LayoutEncoderConverter {

    @Override
    public Properties convert(String layoutPattern) throws Exception {
        return new LogbackLayoutEncoderParser().convert(layoutPattern);
    }
}
